consul/src/main/java/com/linecorp/armeria/internal/consul/ConsulClient.java

```diff
@@ -0,0 +1,291 @@
+    }
+
+    /**
+     * Gets a object mapper.
+     * @return Jackson ObjectMapper
+     */
+    public ObjectMapper getObjectMapper() {
+        return objectMapper;
+    }
+
+    /**
+     * Registers a service to Consul Agent without service ID.
+     *
+     * @param serviceName service name
+     * @param endpoint servicing endpoint
+     * @param check a check for health checking
+     * @return CompletableFuture with registered service ID(auto-generated)
+     */
+    public CompletableFuture<String> register(String serviceName, Endpoint endpoint, @Nullable Check check)
+            throws JsonProcessingException {
+        return AgentServiceClient.of(this)
+                                 .register(serviceName, endpoint.host(), endpoint.port(), check,
+                                           QueryParams.of());
+    }
+
+    /**
+     * Registers a service to Consul Agent with service ID.
+     *
+     * @param serviceId a service ID that identifying a service
+     * @param serviceName a service name to register
+     * @param endpoint an endpoint of service to register
+     * @param check a check for the service
+     * @return CompletableFuture with registered service ID
+     */
+    public CompletableFuture<String> register(String serviceId, String serviceName, Endpoint endpoint,
+                                              Check check) throws JsonProcessingException {
+        return AgentServiceClient.of(this)
+                                 .register(serviceId, serviceName, endpoint.host(), endpoint.port(), check,
+                                           QueryParams.of());
+    }
+
+    /**
+     * De-registers a service to Consul Agent.
+     *
+     * @param serviceId a service ID that identifying a service
+     */
+    public CompletableFuture<Void> deregister(String serviceId) {
+        return AgentServiceClient.of(this).deregister(serviceId);
+    }
+
+    /**
+     * Get registered endpoints with service name from consul agent.
+     */
+    public CompletableFuture<List<Endpoint>> endpoints(String serviceName) {
+        return CatalogClient.of(this)
+                            .endpoints(serviceName, QueryParams.of());
+    }
+
+    /**
+     * Get registered endpoints with service name from consul agent.
+     */
+    public CompletableFuture<List<Endpoint>> healthyEndpoints(String serviceName) {
+        return healthClient().healthyEndpoints(serviceName);
+    }
+
+    /**
+     * Gets a {@code WebClient} for accessing to consul server.
+     */
+    public WebClient consulWebClient() {
+        return webClient;
+    }
+
+    /**
+     * Gets a URL of consul agent.
+     */
+    public String url() {
+        return webClient.uri().toString();
+    }
+
+    private HealthClient healthClient() {
+        if (healthClient == null) {
+            synchronized (ConsulClient.class) {
+                if (healthClient == null) {
+                    healthClient = HealthClient.of(this);
```

Comment:
Please create the `healthClient` in the constructor.

Reply:
Yes.